payments-core/src/main/java/com/stripe/android/payments/bankaccount/ui/CollectBankAccountViewModel.kt

```diff
@@ -8,24 +8,33 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import androidx.savedstate.SavedStateRegistryOwner
 import com.stripe.android.core.Logger
+import com.stripe.android.payments.bankaccount.CollectBankAccountConfiguration
 import com.stripe.android.payments.bankaccount.di.DaggerCollectBankAccountComponent
 import com.stripe.android.payments.bankaccount.domain.AttachLinkAccountSession
 import com.stripe.android.payments.bankaccount.domain.CreateLinkAccountSession
 import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountContract
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountContract.Args.ForPaymentIntent
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountContract.Args.ForSetupIntent
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountResponse
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountResult
+import com.stripe.android.payments.bankaccount.navigation.CollectBankAccountResult.Completed
+import com.stripe.android.payments.bankaccount.ui.CollectBankAccountViewEffect.OpenConnectionsFlow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharedFlow
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-@Suppress("UnusedPrivateMember")
+@Suppress("ConstructorParameterNaming")
 internal class CollectBankAccountViewModel @Inject constructor(
+    // bound instances
     private val args: CollectBankAccountContract.Args,
+    private val _viewEffect: MutableSharedFlow<CollectBankAccountViewEffect>,
```

Comment:
We'll be providing the view events flow at construction time so that we can observe it in tests before viewModel initializes. That way we can test view event emissions that happen on vm#init.

Reply:
- @VisibleForTesting won't help here. Passing the shared flow via constructor allows us to observe **before viewModel initialization**. By the time we want to override the @VisibleForTesting annotated flow, the viewModel will already be initialized and we'll lose that first emission.
- Can you elaborate on why this is weird? inverting dependencies is always a good practice over shadowing them. we don't have a state flow on this screen, but it's a common practice also passing it via constructor params, so that an initial state can be provided on tests to simulate specific scenarios.